awx/main/dispatch/worker/callback.py

```diff
@@ -151,12 +152,14 @@ def flush(self, force=False):
             metrics_singular_events_saved = 0
             metrics_events_batch_save_errors = 0
             metrics_events_broadcast = 0
+            metrics_total_job_event_processing_seconds = datetime.timedelta(seconds=0)
             for cls, events in self.buff.items():
                 logger.debug(f'{cls.__name__}.objects.bulk_create({len(events)})')
                 for e in events:
                     if not e.created:
                         e.created = now
                     e.modified = now
+                    metrics_total_job_event_processing_seconds += e.modified - e.created
```

Comment:
I'd rather this not be saved if the `if not e.created:` above is hit. In that case, you would erroneously record no lag, which isn't theoretically possible. I realize that it would mess up your denominator, so you'd have to track another thing.

Reply:
I dislike the idea that modified could ever be _before_ created, but the intent I definitely get behind.